src/OrbitDB.js

```diff
@@ -58,20 +61,39 @@ let databaseTypes = {
       throw new Error('IPFS is a required argument. See https://github.com/orbitdb/orbit-db/blob/master/API.md#createinstance')
 
     const { id } = await ipfs.id()
-    const directory = options.directory || './orbitdb'
-    const keystore = options.keystore || Keystore.create(path.join(directory, id, '/keystore'))
 
-    const identity = options.identity || await Identities.createIdentity({
-      id: options.id || id,
-      keystore: keystore,
-    })
-    options = Object.assign({}, options, {
-      peerId: id ,
-      directory: directory,
-      keystore: keystore
-    })
-    const orbitdb = new OrbitDB(ipfs, identity, options)
-    return orbitdb
+    if(!options.directory)
+      options.directory = './orbitdb'
+
+    if (!options.storage) {
+      const leveldown = isNode ? require('leveldown') : require('level-js')
+      options.storage = Storage(leveldown)
+      options.storage.preCreate = isNode ? async (directory, options) => {
+        fs.mkdirSync(directory, { recursive: true })
+      } : null;
+    }
+
```

Comment:
Thanks for changing this! 👍 

I now realize there's another, slightly more profound problem here with the browser build. As the check for whether we're in Node or browser happens dynamically (=runtime), webpack can't recognize it (when including OrbitDB as a module, the webpack config for building orbitdb does replace 'leveldown' with 'level-js') and will include both `leveldown` and `level-js` for the user. In [orbit-db-cache](https://github.com/orbitdb/orbit-db-cache), we separated the index/main files between browser and nodejs to solve that. I think we should find a solution for this, possibly similar to what cache does.

You can verify the inclusion by adding `--display-modules` to the webpack options in [build:examples](https://github.com/orbitdb/orbit-db/pull/623/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&owned-by%5B%5D=#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R72) and [build:dist](https://github.com/orbitdb/orbit-db/pull/623/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&owned-by%5B%5D=#diff-b9cfc7f2cdf78a7f4b91a753d10865a2R73) and run them `npm run build:dist |grep level` (observe leveldown is not included) and `npm run npm run build:examples |grep level` (observe leveldown is included).

What do you think?

Reply:
As a side note, it would make since to change https://github.com/orbitdb/orbit-db-storage-adapter/blob/master/src/index.js#L31 to use `await` instead of assuming preCreate is a Promise. That way it works even if the preCreate function is sync or not-a-Promise.

Reply:
I believe this is all set, and the browser versions are building and examples are running locally for me :+1: :tada:

Reply:
Browser examples lgtm locally too :+1: